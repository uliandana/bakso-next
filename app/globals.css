@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --foreground-rgb: 255, 255, 255;
  --background-start-rgb: 188,55,54;
  --background-end-rgb: 188,55,54;
  font-size: calc(768px * 16 / 375);
}

@media (prefers-color-scheme: dark) {
  :root {
    --foreground-rgb: 255, 255, 255;
    --background-start-rgb: 0, 0, 0;
    --background-end-rgb: 0, 0, 0;
  }
}

@layer utilities {
  .card-flip {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(0);
    transition: transform 0.5s ease-in-out;
  }
  .card-flipped {
    position: absolute;
    backface-visibility: hidden;
    transform: rotateY(180deg);
    transition: transform 0.5s ease-in-out;
  }
  .silhouette {
    filter: brightness(0) drop-shadow(0.0625rem 0.125rem 0.25rem rgb(var(--background-start-rgb)));
  }
}

body {
  color: rgb(var(--foreground-rgb));
  background: linear-gradient(
      to bottom,
      transparent,
      rgb(var(--background-end-rgb))
    )
    rgb(var(--background-start-rgb));
}

* {
  animation: fade-in 0.5s ease-in-out;
}

@keyframes loading {
  to { transform: translateX(100%); }
}

@keyframes fade-in {
  from { opacity: 0; }
  to { opacity: 1; }
}

@media only screen and (max-width: 1440px) {
  :root {
    font-size: calc(100vw * 16 / 1440);
  }
}

@media only screen and (max-width: 768px) {
  :root {
    font-size: calc(100vw * 16 / 768);
  }
}
