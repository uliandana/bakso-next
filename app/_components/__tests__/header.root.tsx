import React from 'react';
import { describe, expect, jest, test } from '@jest/globals';
import { createRenderer } from 'react-test-renderer/shallow';
import Header from '../header.root';

describe('./app/_components/header.root', () => {
  test('renders', () => {
    const search = {
      value: 'pika',
      setSearch: jest.fn(),
    };
    const tree = createRenderer().render(<Header search={search} />);
    expect(tree).toMatchSnapshot();
  });

  test('trigger event when input search onchanged', () => {
    const search = {
      value: 'pika',
      setSearch: jest.fn(),
    };
    const res = Header({ search });
    const input = res.props.children[1];
    input.props.onChange({
      target: {
        value: 'mew',
      },
    });
    expect(search.setSearch).toBeCalledWith('mew');
  });
});
