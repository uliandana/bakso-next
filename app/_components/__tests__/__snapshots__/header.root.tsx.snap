// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./app/_components/header.root renders 1`] = `
<header>
  <h1>
    Choose Your Pokemon
  </h1>
  <input
    onChange={[Function]}
    placeholder="Search pokemon"
    type="search"
    value="pika"
  />
</header>
`;
